test/integration/job/job_test.go

```diff
@@ -270,6 +270,158 @@ func TestIndexedJob(t *testing.T) {
+			}
+			if err := waitForEvent(events, job.UID, expectedEventReason); err != nil {
+				t.Errorf("Waiting for event with reason %q: %v", expectedEventReason, err)
+			}
+
+			job, err = clientSet.BatchV1().Jobs(ns.Name).Get(ctx, job.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Failed to get Job: %v", err)
+			}
+			job.Spec.Suspend = pointer.BoolPtr(tc.suspendUpdate)
+			updatedJob, err := clientSet.BatchV1().Jobs(ns.Name).Update(ctx, job, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to update Job: %v", err)
+			}
+			if tc.featureGate && tc.suspendUpdate {
+				expectedActive, expectedStatus, expectedEventReason = 0, v1.ConditionTrue, "Suspended"
+			} else {
+				expectedActive = int(parallelism)
+				if tc.featureGate && tc.suspendCreate {
+					expectedStatus, expectedEventReason = v1.ConditionFalse, "Resumed"
+				} else {
+					expectedStatus, expectedEventReason = "", ""
+				}
+			}
+			validateJobPodsStatus(ctx, t, clientSet, updatedJob, podsByStatus{
+				Active: expectedActive,
+			})
+			job, err = clientSet.BatchV1().Jobs(ns.Name).Get(ctx, job.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Failed to get Job: %v", err)
+			}
+			if got, want := getJobConditionStatus(ctx, job, batchv1.JobSuspended), expectedStatus; got != want {
+				t.Errorf("Unexpected Job condition %q status on update: got %q, want %q", batchv1.JobSuspended, got, want)
+			}
+			if err := waitForEvent(events, job.UID, expectedEventReason); err != nil {
+				t.Errorf("Waiting for event with reason %q: %v", expectedEventReason, err)
+			}
+		})
+	}
+}
+
+func TestSuspendJobControllerRestart(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.SuspendJob, true)()
+
+	closeFn, restConfig, clientSet, ns := setup(t, "suspend")
+	defer closeFn()
+	ctx, cancel := startJobController(restConfig, clientSet)
+	defer func() {
+		cancel()
+	}()
+
+	job, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
+		Spec: batchv1.JobSpec{
+			Parallelism: pointer.Int32Ptr(2),
+			Completions: pointer.Int32Ptr(4),
+			Suspend:     pointer.BoolPtr(true),
+		},
+	})
+	if err != nil {
+		t.Fatalf("Failed to create Job: %v", err)
+	}
+	validateJobPodsStatus(ctx, t, clientSet, job, podsByStatus{
+		Active: 0,
+	})
+
+	// Disable feature gate and restart controller to test that pods get created.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.SuspendJob, false)()
+	cancel()
+	ctx, cancel = startJobController(restConfig, clientSet)
+	job, err = clientSet.BatchV1().Jobs(ns.Name).Get(ctx, job.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get Job: %v", err)
+	}
+	validateJobPodsStatus(ctx, t, clientSet, job, podsByStatus{
+		Active: 2,
+	})
+
+	// Enable feature gate and restart controller to test that pods get deleted.
```

Comment:
Just for the purpose of saving some time, what if we remove this last step? Are we losing too much? I would argue that this is covered in step 1 + the test above.

Reply:
But here we are not testing the changes in apiserver (we are not doing Job updates), only the controller honoring the gate.

Reply:
actually yeah, on second thoughts, I agree. removed.